contracts/actions/common/TakeFlashloan.sol

```diff
@@ -28,7 +32,7 @@ contract TakeFlashloan is Executable, ProxyPermission {
   ServiceRegistry internal immutable registry;
   address internal immutable dai;
 
-  constructor(ServiceRegistry _registry, address _dai) {
+  constructor(ServiceRegistry _registry, address _dai, address _dsGuardFactory) ProxyPermission(_dsGuardFactory) {
```

Comment:
Why not pass the `ServiceRegistry` as a dependency isntead of `dsGuardFactory` address?